### DISCUSSION TITLE
What are the general guidelines of migration and change?

### DISCUSSION THREAD

**Discussion by rocketteering** [jellyfin/jellyfin-web#5243] [General]
I've been thinking of contributing to Jellyfin via code and I was looking around the code for starters. I noticed so, so many packages, many of them filling a same purpose. For example, the project is full of PostCSS, but there's also all Material-UI libraries available? Which one is it that the project is heading to or going to use?

The styling was one issue, but I also just felt the sheer weight of so many packages. Do we have a guideline or and ADR-decision on what we are going to use? Sorry if it's a dumb question, I'm new here after all

**Comment by rocketteering** [comment]
Or more, there's `vite` and tests running with `vitest`. Which is it? Is the project heading out from webpack? Is there any place where these decisions are documented? Or is the even a decision?

**Comment by thornbill** [comment]
Sorry for the delay in responding you are catching me at a really busy time…

I did provide a bit of context in [this previous discussion](https://github.com/jellyfin/jellyfin-web/discussions/5086#discussioncomment-8098140) on some similar questions. As a result of that discussion, [this issue](https://github.com/jellyfin/jellyfin-web/issues/5182) was created to track progress. Feel free to add any questions you have to the comments there also.

We don’t have any formal documentation around decision making although that could certainly be an area of possible improvement. We have _very few_ regular contributors so decisions tend to be somewhat adhoc among the 2-3 of us.

I think it would be easier to track if you ask your questions again in the issue just so we can keep the discussion in one place for discoverability.

Thanks!

**Reply by rocketteering** [reply]
Thanks @thornbill ! I found the issue and it's good to know I'm not the only one pondering the same thing. I'll add my questions there and close this one.
